ads/inabox/inabox-messaging-host.js

```diff
@@ -122,15 +122,25 @@ export class InaboxMessagingHost {
       return false;
     }
 
-    const iframe =
+    const adFrame =
         this.getFrameElement_(message.source, request['sentinel']);
-    if (!iframe) {
+    if (!adFrame) {
       dev().info(TAG, 'Ignored message from untrusted iframe:', message);
       return false;
     }
 
+    const allowedTypes = adFrame.iframe.dataset['ampAllowed'];
+    // escape the type before including it in the regex
+    const type = request['type'].replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
This feels like over engineering... apologies if my outlining options backfired.  Let's go back to the split contains you had before but using \\s for whitespace as what you have here isn't any faster anyway.